src/elicznik/elicznik.py

```diff
@@ -68,3 +71,54 @@ def get_readings(self, start_date, end_date=None):
             (timestamp, float(consumed.get(timestamp)), float(produced.get(timestamp)))
             for timestamp in set(consumed) | set(produced)
         )
+
+    def get_raw_data(self, start_date, end_date=None):
+        end_date = end_date or start_date
+        if start_date == end_date:
+            start_date -= datetime.timedelta(days=1)
+        return self.session.post(
+            self.DATA_URL,
+            data={
+                "form[from]": start_date.strftime("%d.%m.%Y"),
+                "form[to]" : end_date.strftime("%d.%m.%Y"),
+                "form[type]": "godzin", # or "dzien"
+                "form[consum]": 1,
+                "form[oze]": 1,
+                "form[fileType]": "CSV", # or "XLS"
+            },
+        ).content.decode().split('\n')
+
+    def get_data(self, start_date, end_date=None):
+        end_date = end_date or start_date
+        data = csv.reader(self.get_raw_data(start_date, end_date)[1:], delimiter=';')
```

Comment:
Use `csv.DictReader` instead of `csv.reader`.  It will parse the header row automatically and produce a `dict` for each row.